lib/publiccloud/utils.pm

```diff
@@ -885,4 +886,112 @@ sub upload_asset_on_remote {
     $instance->ssh_assert_script_run($mv_cmd);
 }
 
+
+=head2 zypper_call_remote
+
+zypper_call_remote($instance, $cmd, %args)
+
+Remote package management.
+To run on a remote public cloud vm instance a non-interative system manager
+zypper or a transactional-update package command (depending on 'DISTRI'),
+accepting in input the subcommand, then composing the full command to run.
+Free command input also allowed, skipping pre-composition,
+  when 'zypper' or 'transactional-update' already present in input;
+  useful when needing to run a complex one-liner or pipeline package management command.
+
+Parameters:
+  $instance: pointer to publiccloud instance class;
+  $cmd|$args{cmd}: default:'update'; subcommand of zypper or tr-up package management to run
+    OR free full command line, including zypper/tr-up command;
+  %args: dictionary of the arguments defined in input, like: (argument => value, ...):
+    $args{log}: full-path-filename, to save output for upload;
+    $args{global_options}: global options before subcommand, default '-n', as by documentation;
+    $args{environment}: eventual in-line shell pre-setting (i.e. "LANG=C zypper lr ...");
+    $args{proceed_on_failure}: default=0: works as assertion; 1: doesn't stop on error.
+    $args{retry}: to trigger remote retries on error, by default (0) inactve.
+    Included all the $args{...} of run_ssh_command()/retry_ssh_command(), form lib/publiccloud/instance.pm.
+
+Return, same as run_ssh_command(): rc_only=1 [default], as script_run; rc_only=0, as script_output.
+
+=cut
+
+sub zypper_call_remote {
+    my $instance = shift;
+    my %args = testapi::compat_args({cmd => undef}, ['cmd'], @_);
+    return unless ($instance);
+
+    # preset
+    $args{rc_only} //= 1;
+    $args{retry} //= 0;
+    $args{proceed_on_failure} //= 0;
```

Comment:
`run_ssh_command` does not support this parameter in case of timeout it will fail even when this parameter is 1

Reply:
In reality run_ssh_command does suport that parameter only for `script_output` (full standard output), that is when `rc_only = 0`.
Here I extended the usage of this parameter and related effects to the `script_run` case, occurring when we set `rc_only=1`.